pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -318,9 +328,9 @@ func newRevision(set *apps.StatefulSet, revision int64) (*apps.ControllerRevisio
 		revision)
 }
 
-// applyRevision returns a new StatefulSet constructed by restoring the state in revision to set. If the returned error
+// ApplyRevision returns a new StatefulSet constructed by restoring the state in revision to set. If the returned error
 // is nil, the returned StatefulSet is valid.
-func applyRevision(set *apps.StatefulSet, revision *apps.ControllerRevision) (*apps.StatefulSet, error) {
+func ApplyRevision(set *apps.StatefulSet, revision *apps.ControllerRevision) (*apps.StatefulSet, error) {
```

Comment:
@crimsonfaith91 looks like you need it in kubectl/rollback.go, so i don't see another way. perhaps someone else has an idea